pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -1199,6 +1199,11 @@ func isUSBNeeded(c *ConverterContext, vmi *v1.VirtualMachineInstance) bool {
 		return true
 	}
 
+	// USB passthrough is under HostDevices
+	if len(vmi.Spec.Domain.Devices.HostDevices) != 0 {
```

Comment:
Should we check the presence of the devices? Or is it acceptable for any host device?

Reply:
Right, I think we are able to properly check if `CreateHostDevices` from `pkg/virt-launcher/virtwrap/device/hostdevice/usb/hostdev.go` can/did succeed prior to adding the USB controller. I'll fix this.

Reply:
Added this to the code.

```diff
+       usbhostdev "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/device/hostdevice/usb"
...
...
-       // USB passthrough is under HostDevices
-       if len(vmi.Spec.Domain.Devices.HostDevices) != 0 {
+       if usbhostdev.USBDevicesFound(vmi.Spec.Domain.Devices.HostDevices) {
                return true
        }

diff --git a/pkg/virt-launcher/virtwrap/device/hostdevice/usb/hostdev.go b/pkg/virt-launcher/virtwrap/device/hostdevice/usb/hostdev.go
index ddb505809..cf20f549e 100644
--- a/pkg/virt-launcher/virtwrap/device/hostdevice/usb/hostdev.go
+++ b/pkg/virt-launcher/virtwrap/device/hostdevice/usb/hostdev.go
@@ -11,6 +11,16 @@ import (
        "kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
 )

+func USBDevicesFound(vmiHostDevices []v1.HostDevice) bool {
+       for _, device := range vmiHostDevices {
+               env := util.ResourceNameToEnvVar("USB", device.DeviceName)
+               if _, ok := os.LookupEnv(env); ok {
+                       return true
+               }
+       }
+       return false
+}
```